chain/store/store_test.go

```diff
@@ -124,6 +125,51 @@ func TestChainExportImport(t *testing.T) {
 	}
 }
 
+// Test to check if tipset key cids are being stored on snapshot
+func TestChainImportTipsetKeyCid(t *testing.T) {
+
+	ctx := context.Background()
+	cg, err := gen.NewGenerator()
+	require.NoError(t, err)
+
+	buf := new(bytes.Buffer)
+	var last *types.TipSet
+	var tsKeys []types.TipSetKey
+	for i := 0; i < 10; i++ {
+		ts, err := cg.NextTipSet()
+		require.NoError(t, err)
+		last = ts.TipSet.TipSet()
+		tsKeys = append(tsKeys, last.Key())
+	}
+
+	if err := cg.ChainStore().Export(ctx, last, last.Height(), false, buf); err != nil {
+		t.Fatal(err)
+	}
+
+	nbs := blockstore.NewMemorySync()
+	cs := store.NewChainStore(nbs, nbs, datastore.NewMapDatastore(), filcns.Weight, nil)
+	defer cs.Close() //nolint:errcheck
+
+	root, err := cs.Import(ctx, buf)
+	require.NoError(t, err)
+
+	err = cs.SetHead(ctx, last)
```

Comment:
Nit: I would do the sethead _after_ having made the root = last assertion.

Reply:
done